server/tests/delta_test.rs

```diff
@@ -0,0 +1,159 @@
+mod delta_test {
```

Comment:
This is a weird place to put tests in my opinion. Are these very long running or a pain to setup for some reason? Can they not just put alongside the code that they're testing?

Reply:
Feature refresher is already so big, 1700 lines. It is super hard to follow as a newcomer.

Reply:
I think that's a signal that feature refresher is too big and new code needs to be split out, rather than moving tests out

Reply:
Probably add a new delta_refresher.rs, which can still extend FeatureRefresher with methods by using
```rust
impl FeatureRefresher {
  /// The delta code
}

#[cfg(test)]
mod tests {
 /// our delta tests, that way we won't have to compile a separate module either.
}
```

That also means we can readd the #[cfg(test)] to the new method for unleash client.

Reply:
Doing it that way, also allows doing what Simon suggests and add a `delta` compiler flag
basically, in the http module, you'd declare
```rust
#[cfg(feature = "delta")]
mod delta_refresher;
```

Reply:
But also a good observation on the size (I'm assuming the 1700 lines includes the test part of the code as well). 

It might point towards feature_refresher needing to be a folder with multiple files rather than one file.

Reply:
Added it behind compiler flag.